gamemode/cl_bannedmodels.lua

```diff
@@ -1,147 +1,186 @@
-GM.BannedModels = {}
+-- This file provides the functionality for accessing the banned models
+-- menu. It is used locally to determine if a model is viable (aka
+-- whether or not it gets an outline when moused over).
+
+
+GM.BannedModels = {} -- This is used as a hash table where the key is the model string and the value is true.
+local menu
+
 
 function GM:IsModelBanned(model)
-	return table.HasValue(self.BannedModels, model)
+	return self.BannedModels[model] == true
 end
 
+
 function GM:AddBannedModel(model)
-	table.insert(self.BannedModels, model)
+	if self.BannedModels[model] == true then return end -- Prevent duplicates.
```

Comment:
Not sure these checks are really needed. Not much more expensive to always write `true` or `nil` back to this table.